### PR TITLE
[Federation][init-10d] Use the right service names in controller manager.

### PR DESCRIPTION
**Pull request by madhusudancs**
Please review only the last commit here. This is based on PR #36048 which will be reviewed independently.

Design Doc: PR #34484

cc @kubernetes/sig-cluster-federation @nikhiljindal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36294)

### CODE REVIEW

File: federation/pkg/kubefed/init/init.go [hunk 1]
```diff
@@ -501,9 +501,9 @@ func createControllerManager(clientset *client.Clientset, namespace, name, cmNam
 							Command: []string{
 								"/hyperkube",
 								"federation-controller-manager",
-								"--master=https://federation-apiserver",
+								fmt.Sprintf("--master=https://%s", svcName),
 								"--kubeconfig=/etc/federation/controller-manager/kubeconfig",
-								"--dns-provider=gce",
+								"--dns-provider=google-clouddns",
```
Comment (quinton-hoole) [inline federation/pkg/kubefed/init/init.go]: This needs to work for non-Google cloud providers?  e.g. How do I specify that I want to use AWS Route53?

**Comment by dims** [comment]
@quinton-hoole @madhusudancs : looks like this needs a rebase at least. will this make it to v1.5? if not, can you please switch milestone?

**Comment by madhusudancs** [comment]
@dims this is a bug fix for a feature introduced in v1.5, so this has to be in 1.5. I am aware that this needs a rebase and I need to address a review comment. Please give us some time. This is KubeCon+Dev summit week and we are pretty occupied.

**Comment by dims** [comment]
@madhusudancs Ack. This was just a first pass at figuring out what can/should remain on the milestone tag on behalf of the release czar. So thanks for your input. i agree with your reasoning.

File: federation/pkg/kubefed/init/init.go (hunk 1 above)
Comment (madhusudancs) [inline federation/pkg/kubefed/init/init.go]: @quinton-hoole @irfanurrehman fixed this in PR #36492. So I am removing this change from here.

**Review by madhusudancs — COMMENTED** [review]
@quinton-hoole PTAL.

**Comment by madhusudancs** [comment]
Fixed the unit test failure, adding LGTM back.

**Comment by madhusudancs** [comment]
@saad-ali this is a bug fix. Should this go through the exception process?

**Comment by dims** [comment]
@madhusudancs milestone is set, lgtm is applied. so this will go through as you have already mentioned this was a "bug fix". So we are good i think.

**Comment by saad-ali** [comment]
> @saad-ali this is a bug fix. Should this go through the exception process?

Nope, bug fix ok for post-code freeze merge.

**Comment by madhusudancs** [comment]
@dims @saad-ali thanks!

**Pull request merged**